utils_edx_cookiecutters/layered_cookiecutter.py

```diff
@@ -0,0 +1,100 @@
+"""
+    Helps create a cookiecutter using layered approach
+"""
+from os import getenv
+from pathlib import Path
+from shutil import move as shutil_move
+from shutil import rmtree as shutil_rmtree
+
+from cookiecutter.main import cookiecutter
+from edx_lint.cmd.write import write_main
+
+from .run_make import run_make
+
+
+class LayeredCookiecutter():
+    """
+    Class to help facilitate usage of the layered cookiecutter approach
+    """
+
+    # cookiecutter can import a template from either github or from a location on local disk.
+    # If someone is debugging this repository locally, the below block is necessary to pull in
+    #   local versions of the templates
+    EDX_COOKIECUTTER_ROOTDIR = Path(
+        getenv('EDX_COOKIECUTTER_ROOTDIR') or 'https://github.com/edx/edx-cookiecutters.git'
+        )
+
+    def __init__(self, project_rootdir, edx_cookiecutter_rootdir=None):
+        if edx_cookiecutter_rootdir is not None:
+            self.EDX_COOKIECUTTER_ROOTDIR = Path(edx_cookiecutter_rootdir)
+        self.templates = []
+        self.project_rootdir = Path(project_rootdir)
+
+    def move(self, src, dest):
+        """
+        Use to move files or folders without replacement.
+        """
+        src = Path(src)
+        dest = Path(dest)
+        if dest.is_file():
+            src.unlink()
+            return
+        if src.is_dir() and dest.is_dir():
+            for content in src.iterdir():
+                self.move(content, dest / content.name)
+            src.rmdir()
+        else:
+            shutil_move(src, dest)
+
+    def remove(self, path):
+        """
+        Use to remove either a file or a whole directory.
+        """
+        path = Path(path)
+        full_path = self.project_rootdir / path
+        if full_path.is_file():
+            full_path.remove()
+        elif full_path.is_dir():
+            shutil_rmtree(full_path)
+        else:
+            print("{path} not in cookiecutter output".format(path=full_path))
+
+    def add_template(self, template_name, extra_context, remove_object=None):
+        """
+        Used to add templates.
+
+        The templates will be used in order they are added.
+        """
+        if remove_object is None:
+            remove_object = []
+        template_info = {
+                "extra_context": extra_context,
+                "template_name": template_name,
+                "remove_object": remove_object,
+                }
```

Comment:
Excess indentation -- check that pycodestyle etc. is configured properly?
```suggestion
            "extra_context": extra_context,
            "template_name": template_name,
            "remove_object": remove_object,
        }
```